diff --git a/wrapper/target.cpp b/wrapper/target.cpp
index a4f18b5..554479d 100644
--- a/wrapper/target.cpp
+++ b/wrapper/target.cpp
@@ -232,8 +232,8 @@ bool Target::libCXXIsDefaultCXXLib() const {
   if (!OSNum.Num())
     OSNum = getSDKOSNum();

-  return stdlib != libstdcxx && hasLibCXX() && !isGCC() &&
-         OSNum >= OSVersion(10, 9);
+  return (stdlib != libstdcxx && hasLibCXX() && !isGCC() &&
+         OSNum >= OSVersion(10, 9)) || getSDKOSNum() >= OSVersion(10, 14);
 }

 bool Target::isCXX() {
