diff --git a/tools/osxcross_conf.sh b/tools/osxcross_conf.sh
index a0c9571..6cfa79f 100755
--- a/tools/osxcross_conf.sh
+++ b/tools/osxcross_conf.sh
@@ -2,7 +2,12 @@
 
 pushd "${0%/*}" &>/dev/null
 
-OSXCROSS_CONF="../target/bin/osxcross-conf"
+if [ -z "${OC_SYSROOT}" ]; then
+  OSXCROSS_CONF="../target/bin/osxcross-conf"
+else
+  OSXCROSS_CONF="${OC_SYSROOT}/bin/osxcross-conf"
+fi
+
 [ -f $OSXCROSS_CONF ] || { OSXCROSS_CONF=$(which osxcross-conf 2>/dev/null) || exit 1; }
 
 $OSXCROSS_CONF || exit 1
